sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -990,23 +990,30 @@ ssize_t gcoap_encode_link(const coap_resource_t *resource, char *buf,
 
 int gcoap_add_qstring(coap_pkt_t *pdu, const char *key, const char *val)
 {
+    return gcoap_add_qstring2(pdu, key, strlen(key), val, strlen(val));
+}
+
+int gcoap_add_qstring2(coap_pkt_t *pdu, const char *key, size_t key_len, const char *val, size_t val_len)
+{
+    assert(pdu);
+    assert(key);
+    assert(key_len != 0);
```

Comment:
nit-picky: why not simply `assert(key_len)`? I mean otherwise you should also do `assert(pdu != NULL)` and `assert(key != NULL)`, or not 😉